src/utils/route_tracker.py

```
@@ -0,0 +1,79 @@
+            return current_lat, current_lon, i
+        
+        # At a location
+        elif next_arrival <= now <= next_departure:
+            return route[i+1]['latitude'], route[i+1]['longitude'], i + 1
+    
+    # Journey complete
+    last_stop = route[-1]
+    return last_stop['latitude'], last_stop['longitude'], len(route) - 1
+
+if __name__ == "__main__":
+    update_map()
```

Ruff (0.8.2)

Undefined name `update_map`

(F821)